File: go-filmql/structure/Film.go
package structure

import "time"

type Film struct {
	Id int
	FrenchTitle string
	OriginalTitle string
	ReleaseDate time.Time
	Status FilmStatus
	Actors []Actor
	Director Director
}

type Role struct {
	Film Film
	CharacterName string
}

func (film Film) addActor(actor Actor) {
	film.Actors = append(film.Actors, actor)
}

func (film Film) setDirector(director Director) {
	film.Director = director
}

type FilmStatus string

const (
	IN_PROGRESS FilmStatus = "IN_PROGRESS"
	RELEASED    FilmStatus = "RELEASED"
	ABANDONED   FilmStatus = "ABANDONED"
)

File: java-filmql/filmql-metier/src/main/java/org/exemple/filmql/businessobject/Film.java
package org.exemple.filmql.businessobject;

import lombok.*;
import org.exemple.filmql.type.FilmStatus;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

/**
 * A film representation.
 */
@Getter
@RequiredArgsConstructor
public class Film {

  @NonNull private final String id;
  @NonNull private final String frenchTitle;
  @NonNull private final String originalTitle;
  @NonNull private final LocalDate releaseDate;
  @NonNull private final FilmStatus status;
  private final List<Actor> actors = new ArrayList<>();
  @Setter(value = AccessLevel.PACKAGE)
  private Director director;

  /** add an {@link Actor} to the film. */
  public Film withActor(@NonNull Actor actor) {
    actors.add(actor);
    return this;
  }
}
